tests/trainer/test_trainer.py

```diff
@@ -1850,3 +1850,35 @@ def compare_optimizers():
     trainer.max_epochs = 2  # simulate multiple fit calls
     trainer.fit(model)
     compare_optimizers()
+
+
+def test_trainer_predict_verify_config(tmpdir):
+
+    class TestModel(LightningModule):
+
+        def __init__(self):
+            super().__init__()
+            self.layer = torch.nn.Linear(32, 2)
+
+        def forward(self, x):
+            return self.layer(x)
+
+    dataloaders = [torch.utils.data.DataLoader(RandomDataset(32, 2)), torch.utils.data.DataLoader(RandomDataset(32, 2))]
+
+    model = TestModel()
+    datamodule = TestLightningDataModule(dataloaders)
+
+    trainer = Trainer(default_root_dir=tmpdir)
+
+    if datamodule:
+        results = trainer.predict(model, datamodule=datamodule)
+    else:
+        results = trainer.predict(model, dataloaders=dataloaders)
```

Comment:
probably missing some parametrization for case with and without datamodule

Reply:
@Borda Do you mean like this?
```python
@pytest.mark.parametrize('datamodule', [False, True])
```

Reply:
@Borda Updated!

Reply:
Good catch